cmd/lhsm-plugin-az/mover.go

```diff
@@ -210,23 +241,25 @@ func (m *Mover) PrimaryStrToPaths(primaryPath string) ([]string, error) {
 }
 
 func (m *Mover) Archive(ctx context.Context, action dmplugin.Action) error {
+  var sourcePath string
+
   util.Log(pipeline.LogInfo, fmt.Sprintf("%s id:%d archive %s", m.name, action.ID(), action.PrimaryPath()))
 	rate.Mark(1)
 
-  fnames, err := m.PrimaryStrToPaths(action.PrimaryPath())
-	if err != nil {
-		return err
-	}
-
-	if util.ShouldLog(pipeline.LogDebug) {
-		util.Log(pipeline.LogDebug, fmt.Sprintf("Path(s) on FS: %s", strings.Join(fnames, ", ")))
-	}
-
-	if len(fnames) > 1 {
-		util.Log(pipeline.LogDebug, "WARNING: multiple paths returned, using first")
-	}
-	fileID := fnames[0]
-	fileKey := m.destination(fileID)
+  fileKey, err := m.GetFileKey(action.PrimaryPath(), action.FileID())
+  if err != nil {
+    return err
+  }
+
+  // If FileID is explicitly given, then we need to call back into GetFileKey to resolve from FID to path for our source
+  // file.  If it is not given, then we've already done this to resolve fileKey, so don't bother doing it again
+  sourcePath = fileKey
```

Comment:
Would it be worth calling out that this operation is the only one where we need to do this reevaluation to get the source path because it is the only one where the source path is needed?